powerdns/views.py

```diff
@@ -61,6 +61,18 @@ class RecordViewSet(OwnerViewSet):
     filter_fields = ('name', 'type', 'content', 'domain')
     search_fields = filter_fields
 
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        ips = self.request.query_params.getlist('ipaddress')
+        types = self.request.query_params.getlist('type')
+        if ips and types:
+            domain_ids = Record.objects.filter(
+                content__in=ips, type='A').values_list('domain__id', flat=True)
```

Comment:
unfortunately, it will not work. For record A with name `allegro.github.com`, domain is  `github.com`, so here you'd extract all records under `github.com` domain. IMO it should be something similar to https://github.com/allegro/ralph/blob/develop/src/ralph/ui/views/common.py#L725

Flow proposal:
* get all A or PTR records which number is equal to int of ip(s) (this will require calculating `number` value for PTR)
* get all TXT records which `name` is one of selected A records name
* get all CNAME records which content is one of selected A records name

If we don't want to calculate number for PTR, we could just reverse ip addresses (ex. 1.2.3.4 -> 4.3.2.1) and select PTR record with name `<reversed-ip>.in-addr.arpa`